### ISSUE TITLE
Wrong dependency in CMAKE.

### ISSUE THREAD

**Issue by jbrezmorf** [flow123d/flow123d#288]
When I build Flow123D at branch A, then switch to branch B, build it again and then switch back to B,
I experience rebuild of the whole project. However the build trees should be preserved.

Other dependency related issue: after 'git checkout -b' the first build using 'make all' end with error. 
The second build using the same command succeeds.

**Comment by x3mSpeedy** [comment]
I've done some research on this dependency issue. Here're my thoughts.

Most build tools compare the timestamp of the source with the timestamp of the derived file. Meaning rebuild will happen on sources with different timestamps even though content is same.
Git is smart and preserver timestamp on sources which **did not changed**. But if they change its timestamp is set to ```now```.

```state A ``` -> ```switch to B (some files changed)``` -> ```switch back to A  ends up in state A'```
**state A** and **state A'** are similar but different files from branch A and B will have their timestamp changed thus rebuild will occur.

[GitFAQ](https://git.wiki.kernel.org/index.php/GitFaq#Why_isn.27t_Git_preserving_modification_time_on_files.3F)
> Git sets the current time as the timestamp on every file it modifies, but only those. The other files are left untouched, which means build tools will be able to depend on modification time and rebuild properly. If build rules change, that can cause a failure anyway, but that is a far less common problem than accidentally not rebuilding.

So in conclusion switching in similar branches will speed up ```make all``` process. The more branches differs longer rebuild time.

**Comment by jbrezmorf** [comment]
One can use ccache to overcome this problem.  No bug.
